systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaClientsResource.java

```diff
@@ -4,241 +4,53 @@
-                containerBuilder.addNewEnv().withName("CONSUMER_CONFIGURATION").withValue(consumerConfiguration).endEnv();
-
-                String envVariablesSuffix = String.format("_%s", kafkaUserName.replace("-", "_"));
-                containerBuilder.addNewEnv().withName("KAFKA_USER" + envVariablesSuffix).withValue(kafkaUserName).endEnv();
-
-                if (tlsListener) {
-                    if (scramShaUser) {
-                        producerConfiguration += "security.protocol=SASL_SSL\n";
-                        producerConfiguration += saslConfigs(kafkaUser, secretPrefix);
-                        consumerConfiguration += "security.protocol=SASL_SSL\n";
-                        consumerConfiguration += saslConfigs(kafkaUser, secretPrefix);
-                    } else {
-                        producerConfiguration += "security.protocol=SSL\n";
-                        consumerConfiguration += "security.protocol=SSL\n";
-                    }
-                    producerConfiguration +=
-                        SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG + "=/tmp/" + kafkaUserName + "-truststore.p12\n" +
-                            SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG + "=pkcs12\n";
-                    consumerConfiguration +=
-                        SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG + "=/tmp/" + kafkaUserName + "-truststore.p12\n" +
-                            SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG + "=pkcs12\n";
-                } else {
-                    if (scramShaUser) {
-                        producerConfiguration += "security.protocol=SASL_PLAINTEXT\n";
-                        producerConfiguration += saslConfigs(kafkaUser, secretPrefix);
-                        consumerConfiguration += "security.protocol=SASL_PLAINTEXT\n";
-                        consumerConfiguration += saslConfigs(kafkaUser, secretPrefix);
-                    } else {
-                        producerConfiguration += "security.protocol=PLAINTEXT\n";
-                        consumerConfiguration += "security.protocol=PLAINTEXT\n";
-                    }
-                }
-
-                if (tlsUser) {
-                    producerConfiguration +=
-                        SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG + "=/tmp/" + kafkaUserName + "-keystore.p12\n" +
-                            SslConfigs.SSL_KEYSTORE_TYPE_CONFIG + "=pkcs12\n";
-                    consumerConfiguration +=
-                        SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG + "=/tmp/" + kafkaUserName + "-keystore.p12\n" +
-                            SslConfigs.SSL_KEYSTORE_TYPE_CONFIG + "=pkcs12\n";
-
-                    containerBuilder.addNewEnv().withName("PRODUCER_TLS" + envVariablesSuffix).withValue("TRUE").endEnv()
-                        .addNewEnv().withName("CONSUMER_TLS" + envVariablesSuffix).withValue("TRUE").endEnv();
+    @Override
+    public boolean isReady(Deployment resource) {
```

Comment:
Same as other classes - from the name it should be clear whether it returns immediately or wait ... but the code here suggests it waits.